mmeval/metrics/rouge.py

```diff
@@ -0,0 +1,307 @@
+    if pred_len == 0 or reference_len == 0:
+        return dict(precision=0., recall=0., fmeasure=0.)
+    lcs = 0
+    matches = SequenceMatcher(None, pred, reference).get_matching_blocks()
+    for match in matches:
+        lcs += match.size
+    return _get_results(lcs, pred_len, reference_len)
+
+
+def _rougeN_score(pred: Sequence[str], reference: Sequence[str],
+                  n_gram: int) -> Dict[str, float]:
+    """This computes precision, recall and F1 score for the Rouge-N metric.
+
+    Args:
+        pred (Sequence[str]): A predicted sentence.
+        reference (Sequence[str]): A referenced sentence.
+        n_gram (int): The number of words contained in a phrase
+            when calculating word fragments.
+
+    Returns:
+        Dict[str, float]: Calculate the score of rougeN.
+    """
+
+    def _create_ngrams(tokens: Sequence[str], n: int) -> Counter:
+        ngrams: Counter = Counter()
+        for i in range(len(tokens) - n + 1):
+            ngram = tuple(tokens[i:i + n])
+            ngrams[ngram] += 1
+        return ngrams
+
+    pred_ngarms = _create_ngrams(pred, n_gram)
+    reference_ngarms = _create_ngrams(reference, n_gram)
+    pred_len = sum(pred_ngarms.values())
+    reference_len = sum(reference_ngarms.values())
+    if pred_len == 0 or reference_len == 0:
+        return dict(precision=0., recall=0., fmeasure=0.)
+
+    # Take the intersection of n_gram of prediction and reference.
+    hits = sum(
+        min(pred_ngarms[w], reference_ngarms[w]) for w in set(pred_ngarms))
+    return _get_results(hits, pred_len, reference_len)
+
+
+class ROUGE(BaseMetric):
+    """Calculate Rouge Score used for automatic summarization.
+
+    This metric proposed in `ROUGE: A Package for Automatic Evaluation
+    of Summaries <https://www.aclweb.org/anthology/W04-1013.pdf>`_ are
+    common evaluation indicators in the fields of machine translation,
+    automatic summarization, question and answer generation, etc.
+
+    Args:
+         rouge_keys (List or Tuple or int or str):
+            A list of rouge types to calculate.
+            Keys that are allowed are ``L``, and ``1`` through ``9``.
+            Defaults to ``(1, 2, 'L')``.
+        use_stemmer (bool): Use Porter stemmer to strip word
+            suffixes to improve matching. Defaults to False.
+        normalizer (Callable, optional): A user's own normalizer function.
+            If this is ``None``, replacing any non-alpha-numeric characters
+            with spaces is default. Defaults to None.
+        tokenizer (Callable or str, optional): A user's own tokenizer function.
+            Defaults to None.
+        accumulate (str): Useful in case of multi-reference rouge score.
+            ``avg`` takes the average of all references with respect to predictions  # noqa: E501
+            ``best`` takes the best fmeasure score obtained between prediction
+            and multiple corresponding references.
+            Defaults to ``best``.
```

Comment:
```suggestion
        accumulate (str): Useful in case of multi-reference rouge score.
            ``avg`` takes the average of all references with respect to
            predictions. ``best`` takes the best fmeasure score obtained
            between prediction and multiple corresponding references.
            Defaults to ``best``.
```